lib/fastlane/plugin/wpmreleasetoolkit/actions/common/gp_downloadmetadata_action.rb

```diff
@@ -55,19 +55,19 @@ def self.available_options
           FastlaneCore::ConfigItem.new(key: :target_files,
                                        env_name: 'FL_DOWNLOAD_METADATA_TARGET_FILES',
                                        description: 'The hash with the path to the target files and the key to use to extract their content',
-                                       is_string: false),
+                                       type: Hash),
           FastlaneCore::ConfigItem.new(key: :locales,
                                        env_name: 'FL_DOWNLOAD_METADATA_LOCALES',
                                        description: 'The hash with the GlotPress locale and the project locale association',
-                                       is_string: false),
+                                       type: Hash),
```

Comment:
@spencertransier @AliSoftware Is this the correct type for this item? According to [this](https://github.com/wordpress-mobile/release-toolkit/blob/9f2d1920219001ca8613d87155f00ebfd69d9e45/lib/fastlane/plugin/wpmreleasetoolkit/actions/common/gp_downloadmetadata_action.rb#L21) we are checking if it's an `Array` or a `String`. I've also checked the [locales parameter we are passing from WPAndroid](https://github.com/wordpress-mobile/WordPress-Android/blob/fbc44e5df55f064bf7a83602aa65d3aadbff4a4c/fastlane/lanes/localization.rb#L211) and Ruby is telling me that it's an `Array`.

I was trying to test the [auto-retry downloading metadata strings PR](https://github.com/wordpress-mobile/release-toolkit/pull/474) and I've hit an issue where `fastlane` is telling me that `'locales' value must be a Hash! Found Array instead`. I thought it was something I was doing wrong, but I think this change is the culprit.

Reply:
Ah, this is confusing indeed!

> According to [this](https://github.com/wordpress-mobile/release-toolkit/blob/9f2d1920219001ca8613d87155f00ebfd69d9e45/lib/fastlane/plugin/wpmreleasetoolkit/actions/common/gp_downloadmetadata_action.rb#L21) we are checking if it's an Array or a String.

The `params[:locales]` is expected to be an `Enumerable`, on which we call `params[:locales].each do |loc| …` to iterate over each `loc` element. And those individual array elements are what we check for them to be an `Array` or a `String`.

In practice:
 - if `params[:locales]` is a `Hash` (e.g. `{'fr'=>'fr-FR', 'de'=>'de-DE', 'en'=>'en-US'}`, then iterating over that `Hash` yields elements that each are 2-item arrays of the form `[key, value]` (e.g. `['fr', 'fr-FR']`). We thus then falls into the `if loc.is_a?(Array)` branch inside the loop.
 - if `params[:locales]` is an `Array`, then it can be either an `Array` of `Strings` (`['fr', 'de', 'en']`) and fall into the `if loc.is_a?(String)` branch inside the loop… or be an `Array` of 2-element-Arrays (`[['fr', 'fr-FR'], ['de', 'de-DE'], ['en', 'en-US']`) and fall into the `if loc.is_a?(Array)` branch.

So in reality, this parameter can be either a `Hash` or an `Array<String>` or an `Array<[String,String]>` (the latter being what WPAndroid uses) 😓 

I think down the line [our long term plan was to replace this legacy `gp_download_metadata` with a more modern API](pdnsEh-jJ-p2), including because of this odd API around the parameters like `locales` here. But until then we're stuck with this API that is a bit confusing and too flexible to be consistent.

---

I guess the best option here would be to set `type: Enumerable` and better document the parameter in the `description`. That being said, I've never seen `type: Enumerable` being used for a `ConfigItem` (as it's more common to provide explicit types), so we better test that this more generic type would work and be supported by `ConfigItem`'s implementation. So I guess this would be the ideal time to write some specs for that action (even minimalistic ones for now, that would only focus on feeding different type of inputs for this parameter) to ensure `type: Enumerable` would work as expected (or what would need to be used instead if not).

Reply:
Could we temporarily revert the change and open a separate issue for this? Since it's a breaking issue we need to address it immediately and I unfortunately don't have the bandwidth to handle it properly.

I could also test if `type: Enumerable` works for WPAndroid and handle the hotfix and the releases if that's good enough for now.

cc @spencertransier In case you have more availability to follow up on this better than I can.

Reply:
@oguzkocer I'll revert it and get the 7.1.2 release going.

Reply:
Created separate issue for this here: https://github.com/wordpress-mobile/release-toolkit/issues/484